### PR TITLE
DT-2148 Introduce ROLE_PRISONER_SEARCH role for endpoints used by client credentials

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerMatchResource.kt [hunk 1]
```diff
@@ -23,16 +23,16 @@ import javax.validation.Valid
 @RestController
 @Validated
 @RequestMapping(value = ["/match-prisoners"], produces = [MediaType.APPLICATION_JSON_VALUE])
-@PreAuthorize("hasRole('ROLE_GLOBAL_SEARCH')")
+@PreAuthorize("hasAnyRole('GLOBAL_SEARCH','ROLE_PRISONER_SEARCH')")
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerMatchResource.kt:26]: missing `ROLE_` prefix?

Comment (petergphillips) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerMatchResource.kt:26]: looks like the tests should that it works for both, but for consistency worth just using `ROLE_GLOBAL_SEARCH` everywhere instead

**Review by petergphillips — DISMISSED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerMatchResource.kt (hunk 1 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerMatchResource.kt]: updated

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**